src/crystal/system/unix/file.cr

```diff
@@ -0,0 +1,236 @@
+        m = LibC::O_WRONLY
+        o = LibC::O_CREAT | LibC::O_APPEND
+      else
+        raise "Invalid access mode #{mode}"
+      end
+
+      case mode.size
+      when 1
+        # Nothing
+      when 2
+        case mode[1]
+        when '+'
+          m = LibC::O_RDWR
+        when 'b'
+          # Nothing
+        else
+          raise "Invalid access mode #{mode}"
+        end
+      else
+        raise "Invalid access mode #{mode}"
+      end
+
+      oflag = m | o
+    end
+
+    def mktemp(name, extension)
+      tmpdir = tempdir + ::File::SEPARATOR
+      path = "#{tmpdir}#{name}.XXXXXX#{extension}"
+
+      if extension
+        fd = LibC.mkstemps(path, extension.bytesize)
+      else
+        fd = LibC.mkstemp(path)
+      end
+
+      raise Errno.new("mkstemp") if fd == -1
+      {fd, path}
+    end
+
+    def tempdir
+      tmpdir = ENV["TMPDIR"]? || "/tmp"
+      tmpdir.rchop(::File::SEPARATOR)
+    end
+
+    def stat(path)
+      if LibC.stat(path.check_no_null_byte, out stat) != 0
+        raise Errno.new("Unable to get stat for '#{path}'")
+      end
+      ::File::Stat.new(stat)
+    end
+
+    def lstat(path)
+      if LibC.lstat(path.check_no_null_byte, out stat) != 0
+        raise Errno.new("Unable to get lstat for '#{path}'")
+      end
+      ::File::Stat.new(stat)
+    end
+
+    def empty?(path)
+      begin
+        stat(path).size == 0
+      rescue Errno
+        raise Errno.new("Error determining size of '#{path}'")
+      end
+    end
+
+    def exists?(path)
+      accessible?(path, LibC::F_OK)
+    end
+
+    def readable?(path) : Bool
+      accessible?(path, LibC::R_OK)
+    end
+
+    def writable?(path) : Bool
+      accessible?(path, LibC::W_OK)
+    end
+
+    def executable?(path) : Bool
+      accessible?(path, LibC::X_OK)
+    end
+
+    private def accessible?(path, flag)
+      LibC.access(path.check_no_null_byte, flag) == 0
+    end
+
+    def file?(path) : Bool
+      if LibC.stat(path.check_no_null_byte, out stat) != 0
+        if Errno.value == Errno::ENOENT
+          return false
+        else
+          raise Errno.new("stat")
+        end
+      end
+      ::File::Stat.new(stat).file?
+    end
+
+    def chown(path, uid : Int, gid : Int, follow_symlinks)
+      ret = if !follow_symlinks && symlink?(path)
+              LibC.lchown(path, uid, gid)
+            else
+              LibC.chown(path, uid, gid)
+            end
+      raise Errno.new("Error changing owner of '#{path}'") if ret == -1
+    end
+
+    def chmod(path, mode : Int)
+      if LibC.chmod(path, mode) == -1
+        raise Errno.new("Error changing permissions of '#{path}'")
+      end
+    end
+
+    def delete(path)
+      err = LibC.unlink(path.check_no_null_byte)
+      if err == -1
+        raise Errno.new("Error deleting file '#{path}'")
+      end
+    end
+
+    def real_path(path)
+      real_path_ptr = LibC.realpath(path, nil)
+      raise Errno.new("Error resolving real path of #{path}") unless real_path_ptr
```

Comment:
Same `path.inspect`, this and following error messages omit the single-quote. (Consistency)

Reply:
ditto